zbus/src/proxy/mod.rs

```diff
@@ -1018,6 +1018,49 @@ impl<'a> Proxy<'a> {
             name: self.destination().clone(),
         })
     }
+
+    /// Wait for the property `name` to meet the `predicate`.
+    ///
+    /// If the property already meets the predicate, this function will return immediately.
+    ///
+    /// If the property is cachable, this function will poll the property stream until the predicate
+    /// is met. If the property is uncacheable, this function will busy await until the predicate is
+    /// met. (not recommended)
+    ///
+    /// Note that zbus doesn't queue the updates. If the listener is slower than the receiver, it
+    /// will only receive the last update.
+    pub async fn wait_property_for<T>(
```

Comment:
I'm now wondering, isn't this achieving the same as you could with [futures API](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.filter) easily? 🤔

Reply:
It's a convenience function of
```rust
use futures::{FutureExt as _, StreamExt as _};

let fut = {
    let event_stream = (proxy.receive_property_changed(name).await)
        .then(|ev| async move { ev.get().await });
    let mut stream = std::pin::pin!(
        futures::stream::once(proxy.get_property::<T>(name).await)
            .chain(event_stream) // impl Stream<Item = zbus::Result<T>>
            .filter(predicate)
    );
    // the future roughly equivalent to `proxy.wait_property_for(name, predicate)`
    stream.next().map(|item| item.unwrap()) // impl Future<Output = zbus::Result<T>>
};
```